src/main/java/net/coderbot/iris/gui/screen/ShaderPackScreen.java

```diff
@@ -0,0 +1,147 @@
+			this.shaderPacks.method_31322(false);
+		}
+		this.children.add(shaderPacks);
+
+		this.addButton(new ButtonWidget(bottomCenter + 104, this.height - 27, 100, 20, ScreenTexts.DONE, button -> {
+			applyChanges();
+			onClose();
+		}));
+		this.addButton(new ButtonWidget(bottomCenter, this.height - 27, 100, 20, new TranslatableText("options.iris.apply"), button -> this.applyChanges()));
+		this.addButton(new ButtonWidget(bottomCenter - 104, this.height - 27, 100, 20, ScreenTexts.CANCEL, button -> this.onClose()));
+		this.addButton(new ButtonWidget(topCenter - 78, this.height - 51, 152, 20, new TranslatableText("options.iris.openShaderPackFolder"), button -> Util.getOperatingSystem().open(Iris.shaderpacksDirectory.toFile())));
+		this.addButton(new ButtonWidget(topCenter + 78, this.height - 51, 152, 20, new TranslatableText("options.iris.refreshShaderPacks"), button -> this.shaderPacks.refresh()));
+	}
+
+	@Override
+	public void tick() {
+		super.tick();
+		if (this.addedPackDialogTimer > 0) {
+			this.addedPackDialogTimer--;
+		}
+	}
+
+	@Override
+	public void filesDragged(List<Path> paths) {
+		List<Path> packs = paths.stream().filter(Iris::isValidShaderpack).collect(Collectors.toList());
+		for (Path pack : packs) {
+			if (Iris.isValidShaderpack(pack)) {
+				String fileName = pack.getFileName().toString();
+				try {
+					Files.copy(pack, Iris.shaderpacksDirectory.resolve(fileName));
+				} catch (IOException e) {
+					e.printStackTrace();
+					this.addedPackDialog = new TranslatableText(
+							"options.iris.shaderPackSelection.copyError",
+							fileName
+					).formatted(Formatting.ITALIC, Formatting.RED);
+					this.addedPackDialogTimer = 100;
+					this.shaderPacks.refresh();
+					return;
+				}
+			}
+		}
+		if (packs.size() > 0) {
+			if (packs.size() == 1) {
+				this.addedPackDialog = new TranslatableText(
+						"options.iris.shaderPackSelection.addedPack",
+						packs.get(0).getFileName().toString()
+				).formatted(Formatting.ITALIC, Formatting.YELLOW);
+			} else {
+				this.addedPackDialog = new TranslatableText(
+						"options.iris.shaderPackSelection.addedPacks",
+						packs.size()
+				).formatted(Formatting.ITALIC, Formatting.YELLOW);
+			}
+		} else {
+			this.addedPackDialog = new TranslatableText(
+					"options.iris.shaderPackSelection.failedAdd"
+			).formatted(Formatting.ITALIC, Formatting.RED);
+		}
+		this.addedPackDialogTimer = 100;
+		this.shaderPacks.refresh();
+	}
+
+	@Override
+	public void onClose() {
+		this.client.openScreen(parent);
+	}
+
+	private void applyChanges() {
+		ShaderPackListWidget.BaseEntry base = this.shaderPacks.getSelected();
+		if (!(base instanceof ShaderPackListWidget.ShaderPackEntry)) {
+			return;
+		}
+		ShaderPackListWidget.ShaderPackEntry entry = (ShaderPackListWidget.ShaderPackEntry)base;
+		String name = entry.getPackName();
+		Iris.getIrisConfig().setShaderPackName(name);
+		try {
+			Iris.reload();
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
Maybe use `Iris.logger` to make it more clear when this happens?

Reply:
Will do.